docs/topics/request-response.rst

```diff
@@ -205,6 +205,12 @@ Request objects
         ``failure.request.cb_kwargs`` in the request's errback. For more information,
         see :ref:`errback-cb_kwargs`.
 
+    .. attribute:: Request.attributes
+
+        A tuple of :class:`str` objects containing all public attribute names (and ``__init__`` parameters)
+        for the class. Currently used by :meth:`Request.replace`, :meth:`Request.to_dict` and
+        :func:`~scrapy.utils.request.request_from_dict`.
```

Comment:
The current text to me sounds like it’s a superset of attribute names and `__init__` parameters (i.e. a parameter that is only available as one of those would be in `attributes`). Assuming I’m not getting the implementation wrong, what about:

```suggestion
        A tuple of :class:`str` objects containing the name of all public 
        attributes of the class that are also keyword parameters of the 
        ``__init__`` method.
        
        Currently used by :meth:`Request.replace`, :meth:`Request.to_dict` and
        :func:`~scrapy.utils.request.request_from_dict`.
```

I also wonder if it would not be better to make this a docstring and use `.. autodata` or similar here.

Reply:
Nice, I didn't know attribute docstrings were a thing, thanks.